tests/fixture/tmpnet/monitor_processes.go

```diff
@@ -0,0 +1,496 @@
+
+	return startCollector(ctx, log, cmdName, args, config)
+}
+
+// startPromtail ensures a promtail process is running to collect logs from local nodes.
+func startPromtail(ctx context.Context, log logging.Logger) error {
+	cmdName := promtailCmd
+
+	args := fmt.Sprintf("-config.file=%s.yaml", cmdName)
+
+	username, password, err := getCollectorCredentials(cmdName)
+	if err != nil {
+		return err
+	}
+
+	workingDir, err := getWorkingDir(cmdName)
+	if err != nil {
+		return err
+	}
+
+	serviceDiscoveryDir, err := getServiceDiscoveryDir(cmdName)
+	if err != nil {
+		return err
+	}
+
+	config := fmt.Sprintf(`
+server:
+  http_listen_port: %d
+  grpc_listen_port: 0
+
+positions:
+  filename: %s/positions.yaml
+
+client:
+  url: "%s/api/prom/push"
+  basic_auth:
+    username: "%s"
+    password: "%s"
+
+scrape_configs:
+  - job_name: "avalanchego"
+    file_sd_configs:
+      - files:
+          - '%s/*.json'
+`, promtailHTTPPort, workingDir, getLokiURL(), username, password, serviceDiscoveryDir)
+
+	return startCollector(ctx, log, cmdName, args, config)
+}
+
+func getWorkingDir(cmdName string) (string, error) {
+	tmpnetDir, err := getTmpnetPath()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(tmpnetDir, cmdName), nil
+}
+
+func getServiceDiscoveryDir(cmdName string) (string, error) {
+	tmpnetDir, err := getTmpnetPath()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(tmpnetDir, cmdName, "file_sd_configs"), nil
+}
+
+func getPIDPath(workingDir string) string {
+	return filepath.Join(workingDir, "run.pid")
+}
+
+// startCollector starts a collector process if it is not already running.
+func startCollector(
+	ctx context.Context,
+	log logging.Logger,
+	cmdName string,
+	args string,
+	config string,
+) error {
+	// Determine paths
+	workingDir, err := getWorkingDir(cmdName)
+	if err != nil {
+		return err
+	}
+	pidPath := getPIDPath(workingDir)
+
+	// Ensure required paths exist
+	if err := os.MkdirAll(workingDir, perms.ReadWriteExecute); err != nil {
+		return fmt.Errorf("failed to create %s dir: %w", cmdName, err)
+	}
+	if err := os.MkdirAll(filepath.Join(workingDir, "file_sd_configs"), perms.ReadWriteExecute); err != nil {
+		return fmt.Errorf("failed to create %s file_sd_configs dir: %w", cmdName, err)
+	}
+
+	// Check if the process is already running
+	if process, err := processFromPIDFile(cmdName, pidPath); err != nil {
+		return err
+	} else if process != nil {
+		log.Info(cmdName + " is already running")
+		return nil
+	}
+
+	// Clear any stale pid file
+	if err := clearStalePIDFile(log, cmdName, pidPath); err != nil {
+		return err
+	}
+
+	// Check if the specified command is available in the path
+	if _, err := exec.LookPath(cmdName); err != nil {
+		return fmt.Errorf("%s command not found. Maybe run 'nix develop'?", cmdName)
+	}
+
+	// Write the collector config file
+	confFilename := cmdName + ".yaml"
+	confPath := filepath.Join(workingDir, confFilename)
+	log.Info("writing "+cmdName+" config",
```

Comment:
I think it's generally frowned upon to have a dynamic log message with zap (and that fields should generally be used instead).

Although it doesn't seem too crazy here - so 🤷